### PR TITLE
Feature/plugins and rtl

### PR DESCRIPTION
**Pull request by chrisguida**
This release contains:
  - Update to v0.10.1
  - Enable summary plugin
  - Enable rebalance plugin
  - Add c-lightning-REST plugin for RTL compatibility
  - Enable accepting of dual fund requests

Update 2021-08-27: 
Todo:

Allow disable plugins from config
Output macaroon in stats

Update 2021-08-31: All TODOs done

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -67,16 +67,36 @@ RUN ./configure
...
+RUN chmod a+x /usr/local/libexec/c-lightning/plugins/rebalance/rebalance.py
+
+#summary
+ADD ./plugins/summary /usr/local/libexec/c-lightning/plugins/summary
+RUN pip3 install -r /usr/local/libexec/c-lightning/plugins/summary/requirements.txt
+RUN chmod a+x /usr/local/libexec/c-lightning/plugins/summary/summary.py
+
+#c-lightning-REST
+ADD ./c-lightning-REST /usr/local/libexec/c-lightning/plugins/c-lightning-REST
+# ADD /usr/src/app/node_modules/ /usr/local/libexec/c-lightning/plugins/c-lightning-REST/node_modules
```
Comment (ProofOfKeags) [inline Dockerfile:92]:
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -67,16 +67,36 @@ RUN ./configure
...
+ADD ./plugins/summary /usr/local/libexec/c-lightning/plugins/summary
+RUN pip3 install -r /usr/local/libexec/c-lightning/plugins/summary/requirements.txt
+RUN chmod a+x /usr/local/libexec/c-lightning/plugins/summary/summary.py
+
+#c-lightning-REST
+ADD ./c-lightning-REST /usr/local/libexec/c-lightning/plugins/c-lightning-REST
+# ADD /usr/src/app/node_modules/ /usr/local/libexec/c-lightning/plugins/c-lightning-REST/node_modules 
+WORKDIR /usr/local/libexec/c-lightning/plugins/c-lightning-REST
+RUN npm install --only=production
+RUN npm audit fix
```
Comment (ProofOfKeags) [inline Dockerfile:95]: I'm not sure this is a good idea.

File: configurator/src/config.template [hunk 3]
```diff
@@ -4,6 +4,12 @@ bitcoin-rpcpassword={bitcoin_rpc_pass}
 bitcoin-rpcconnect={bitcoin_rpc_host}
 bitcoin-rpcport={bitcoin_rpc_port}
 plugin=/usr/local/libexec/c-lightning/plugins/c-lightning-http-plugin
+plugin=/usr/local/libexec/c-lightning/plugins/rebalance/rebalance.py
+plugin=/usr/local/libexec/c-lightning/plugins/summary/summary.py
+plugin=/usr/local/libexec/c-lightning/plugins/c-lightning-REST/plugin.js
```
Comment (ProofOfKeags) [inline configurator/src/config.template:9]: does this mean they are all turned on by default?

File: configurator/src/config.template [hunk 4]
```diff
@@ -4,6 +4,12 @@ bitcoin-rpcpassword={bitcoin_rpc_pass}
...
+rest-port=3001
+rest-docport=4001
+rest-protocol=https
```
Comment (ProofOfKeags) [inline configurator/src/config.template:12]: this is listening presumably on a different interface than the original rest plugin on line 6, should we deprecate the one on line 6, would that destroy spark's ability to work?

File: configurator/src/main.rs [hunk 5]
```diff
@@ -280,6 +282,16 @@ fn main() -> Result<(), anyhow::Error> {
         }
     }
 
+    let macaroon_path = Path::new("/usr/local/libexec/c-lightning/plugins/c-lightning-REST/certs/access.macaroon");
```
[the remark this answers is not in the document]
Comment (chrisguida) [reply configurator/src/main.rs]: Done

File: configurator/src/config.template (hunk 3 above)
Comment (chrisguida) [reply configurator/src/config.template]: Done.

File: configurator/src/config.template (hunk 4 above)
Comment (chrisguida) [reply configurator/src/config.template]: Leaving both plugins in for now.

**Comment by chrisguida** [comment]
Let's leave it for now, it can be rebased once the 030 stuff goes to master

**Comment by chrisguida** [comment]
Closing in favor of https://github.com/Start9Labs/c-lightning-wrapper/pull/21

**Pull request closed without merging**